x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -0,0 +1,215 @@
+    try {
+      // first search for the alert by id, then use the alert info to check if user has access to it
+      const alert = await this.fetchAlert({
+        id,
+        index,
+      });
+
+      // this.authorization leverages the alerting plugin's authorization
+      // client exposed to us for reuse
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: alert[RULE_ID],
+        consumer: alert[OWNER],
+        operation: ReadOperations.Get,
+        entity: AlertingAuthorizationEntity.Alert,
+      });
+
+      this.auditLogger?.log(
+        alertAuditEvent({
+          action: AlertAuditAction.GET,
+          id,
+        })
+      );
+
+      return alert;
+    } catch (error) {
+      this.logger.debug(`Error fetching alert with id of "${id}"`);
+      this.auditLogger?.log(
+        alertAuditEvent({
+          action: AlertAuditAction.GET,
+          id,
+          error,
+        })
+      );
+      throw error;
+    }
+  }
+
+  public async update<Params extends AlertTypeParams = never>({
+    id,
+    data,
+    index,
+  }: UpdateOptions<Params>): Promise<ParsedTechnicalFields | null | undefined> {
+    try {
+      const alert = await this.fetchAlert({
+        id,
+        index,
+      });
+
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: alert[RULE_ID],
+        consumer: alert[OWNER],
+        operation: WriteOperations.Update,
+        entity: AlertingAuthorizationEntity.Alert,
+      });
+
+      const updateParameters = {
+        id,
+        index,
+        body: {
+          doc: {
+            [ALERT_STATUS]: data.status,
+          },
+        },
+      };
+
+      const res = await this.esClient.update<ParsedTechnicalFields, unknown, unknown, unknown>(
+        updateParameters
+      );
+
+      this.auditLogger?.log(
+        alertAuditEvent({
+          action: AlertAuditAction.UPDATE,
+          id,
+        })
+      );
+
+      return res.body.get?._source;
+    } catch (error) {
+      this.auditLogger?.log(
+        alertAuditEvent({
+          action: AlertAuditAction.UPDATE,
+          id,
+          error,
+        })
+      );
+      throw error;
+    }
+  }
+
+  public async getAuthorizedAlertsIndices(featureIds: string[]): Promise<string[] | undefined> {
+    const augmentedRuleTypes = await this.authorization.getAugmentRuleTypesWithAuthorization(
+      featureIds
+    );
+
+    const arrayOfAuthorizedRuleTypes = Array.from(augmentedRuleTypes.authorizedRuleTypes);
+
+    // As long as the user can read a minimum of one type of rule type produced by the provided feature,
+    // the user should be provided that features' alerts index.
+    // Limiting which alerts that user can read on that index will be done via the findAuthorizationFilter
+    const authorizedFeatures = arrayOfAuthorizedRuleTypes.reduce(
+      (acc, ruleType) => acc.add(ruleType.producer),
+      new Set<string>()
+    );
```

Comment:
optional: it seems like you can simplify this a bit:

```suggestion
    const augmentedRuleTypes = await this.authorization.getAugmentRuleTypesWithAuthorization(
      featureIds
    );

    // As long as the user can read a minimum of one type of rule type produced by the provided feature,
    // the user should be provided that features' alerts index.
    // Limiting which alerts that user can read on that index will be done via the findAuthorizationFilter
    const authorizedFeatures = new Set();
    for (const ruleType of augmentedRuleTypes.authorizedRuleTypes) {
      authorizedFeatures.add(ruleType.producer);
    }
```

Reply:
Thanks!

Reply:
resolved here: 6111991397dd51f51afab7c6a2cda30db267b357